src/NUnitTestAdapter/NUnitEventListener.cs

```diff
@@ -36,99 +37,28 @@ public NUnitEventListener(ITestExecutionRecorder testLog, TestConverter testConv
             this.testConverter = testConverter;
         }
 
-        public void RunStarted(string name, int testCount)
-        {
-            testLog.SendMessage(TestMessageLevel.Informational, "Run started: " + name);
-        }
-
-        public void RunFinished(Exception exception)
-        {
-        }
-
-        public void RunFinished(NUnit.Core.TestResult result)
-        {
-        }
-
-        public string Output { get; private set; }
+        #region ITestEventListener
 
-        public void SuiteStarted(TestName testName)
+        public void OnTestEvent(string report)
         {
-
-        }
-
-        public void SuiteFinished(NUnit.Core.TestResult result)
-        {
-            if ((result.IsError || result.IsFailure) &&
-                (result.FailureSite == FailureSite.SetUp || result.FailureSite == FailureSite.TearDown))
+            var node = XmlHelper.CreateXmlNode(report);
```

Comment:
I was of too minds with things like this. The call is made by the NUnit framework, which we might decide to trust. OTOH, a Guard clause or some other check may be a good idea.